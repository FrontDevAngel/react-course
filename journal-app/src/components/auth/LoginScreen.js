import React from 'react';
import { Link } from 'react-router-dom';

export const LoginScreen = () => {
    return (
        <div className='auth__container'>
            <div className='auth__col auth__col--intro'>
                <h1 className='auth__title'>Login Screen</h1>
                <h3 className='auth__subtitle'>Inicios de sesión recientes</h3>
                <p>Haz click en tu foto o agregar una cuenta</p>
                <div className='auth__cards'></div>
            </div>
            <div className='auth__col auth__col--form'>
                <form className='form form--auth'>
                    <input
                        type='text'
                        name='email'
                        required
                        autoFocus='1'
                        className='form__input form__input--text'
                        placeholder='Correo electrónico o número de teléfono'
                        aria-label='Correo electrónico o número de teléfono'
                    />
                    <input
                        type='password'
                        name='password'
                        required
                        className='form__input form__input--password'
                        placeholder='Contraseña'
                        aria-label='Contraseña'
                    />
                    <button className='form__submit' type='submit' name='login'>
                        Iniciar Sesión
                    </button>
                    <Link className='form__link' to='/auth/register'>
                        Crear una cuenta
                    </Link>
                </form>
            </div>
        </div>
    );
};
