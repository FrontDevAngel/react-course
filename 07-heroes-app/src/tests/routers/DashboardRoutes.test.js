import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';

describe('Pruebas a <DashboardRoutes />', () => {
    test('Debe mostrarse correctamente ', () => {
        const contextValue = {
            user: { logged: false },
            dispatch: jest.fn(),
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrarse el nombre del usuario si esta autenticado', () => {
        const contextValue = {
            user: { logged: false, name: 'Juanito' },
            dispatch: jest.fn(),
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>,
        );

        expect(wrapper.find('.text-info').text().trim()).toBe('Juanito');
    });
});
