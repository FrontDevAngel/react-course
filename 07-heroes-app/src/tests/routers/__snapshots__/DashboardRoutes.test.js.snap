// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <DashboardRoutes /> Debe mostrarse correctamente  1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark px-4"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        />
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container my-5"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="card-columns animate__animated animate__fadeIn"
      >
        <div
          className="card"
        >
          <img
            alt="Spider Man"
            className="card-img-top"
            src="./assets/heroes/marvel-spider.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Spider Man
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Peter Parker
            </h6>
            <p
              className="card-text"
            >
              Amazing Fantasy #15
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-spider"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Captain America"
            className="card-img-top"
            src="./assets/heroes/marvel-captain.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Captain America
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Steve Rogers
            </h6>
            <p
              className="card-text"
            >
              Captain America Comics #1
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-captain"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Iron Man"
            className="card-img-top"
            src="./assets/heroes/marvel-iron.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Iron Man
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Tony Stark
            </h6>
            <p
              className="card-text"
            >
              Tales of Suspense #39
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-iron"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Thor"
            className="card-img-top"
            src="./assets/heroes/marvel-thor.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Thor
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Thor Odinson
            </h6>
            <p
              className="card-text"
            >
              Journey into Myster #83
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-thor"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Hulk"
            className="card-img-top"
            src="./assets/heroes/marvel-hulk.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hulk
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Bruce Banner
            </h6>
            <p
              className="card-text"
            >
              The Incredible Hulk #1
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-hulk"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Wolverine"
            className="card-img-top"
            src="./assets/heroes/marvel-wolverine.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Wolverine
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              James Howlett
            </h6>
            <p
              className="card-text"
            >
              The Incredible Hulk #180
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-wolverine"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Daredevil"
            className="card-img-top"
            src="./assets/heroes/marvel-daredevil.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Daredevil
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Matthew Michael Murdock
            </h6>
            <p
              className="card-text"
            >
              Daredevil #1
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-daredevil"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Hawkeye"
            className="card-img-top"
            src="./assets/heroes/marvel-hawkeye.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hawkeye
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Clinton Francis Barton
            </h6>
            <p
              className="card-text"
            >
              Tales of Suspense #57
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-hawkeye"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Cyclops"
            className="card-img-top"
            src="./assets/heroes/marvel-cyclops.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Cyclops
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Scott Summers
            </h6>
            <p
              className="card-text"
            >
              X-Men #1
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-cyclops"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <img
            alt="Silver Surfer"
            className="card-img-top"
            src="./assets/heroes/marvel-silver.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Silver Surfer
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Norrin Radd
            </h6>
            <p
              className="card-text"
            >
              The Fantastic Four #48
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
            </p>
            <a
              href="/hero/marvel-silver"
              onClick={[Function]}
            >
              Ver más...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
